hack/generate-authors.sh

```diff
@@ -1,15 +1,19 @@
 #!/usr/bin/env bash
+
 set -e
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")/.."
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOTDIR="$(git -C "$SCRIPTDIR" rev-parse --show-toplevel)"
+
+# ensure that the sort order is not locale-dependent
+export LC_ALL=C.UTF-8 
+
+set -x
 
 # see also ".mailmap" for how email addresses and names are deduplicated
+tee "${ROOTDIR}/AUTHORS" <<-EOF
+	# This file is @generated and lists all contributors to the repository.
+	# See hack/generate-authors.sh to make modifications.
 
```

Comment:
Curioius; is `@generated` a specific keyword here?

If it's not, perhaps (as most of the repository follows Go conventions) we could consider using Go's format for this; https://github.com/golang/go/issues/41196, e.g. something like;

```suggestion
	# Code generated by hack/generate-authors.sh. DO NOT EDIT.
	# This file lists all contributors to the repository. See
	# hack/generate-authors.sh to make modifications.
```

Reply:
https://generated.at/

We could combine the two if you like.

Reply:
TIL about that; wondering if we should write a proposal for go to also accept `Code @generated by ...` as prefix

Reply:
It's funny that https://generated.at does not have a (visible) link to a GitHub / GitLab repository, is it s a privately maintained project?

Reply:
As I recall, it's privately maintained by a (now former? :laughing:) Facebook engineer. I was made aware of it in my capacity as maintainer of another large open source project several years ago, so I'm not sure.